src/createPackage/models/createPackageManager.ts

```diff
@@ -12,50 +16,90 @@ import {
   IWorkerInput,
   JobDuplicationParams,
   IJobParameters,
-  ICallbackResposne,
+  ICallbackResposne as ICallbackResponse,
   JobResponse,
+  MergerSourceType,
+  IMapSource,
+  ICallbackTarget,
 } from '../../common/interfaces';
 import { JobManagerWrapper } from '../../clients/jobManagerWrapper';
 
 @injectable()
 export class CreatePackageManager {
+  private readonly tilesProvider: MergerSourceType;
   public constructor(
     @inject(SERVICES.LOGGER) private readonly logger: Logger,
     @inject(JobManagerWrapper) private readonly jobManagerClient: JobManagerWrapper,
     @inject(RasterCatalogManagerClient) private readonly rasterCatalogManager: RasterCatalogManagerClient
-  ) {}
+  ) {
+    this.tilesProvider = config.get('tilesProvider');
+    this.tilesProvider = this.tilesProvider.toUpperCase() as MergerSourceType;
+  }
 
-  public async createPackage(userInput: ICreatePackage): Promise<ICreateJobResponse | ICallbackResposne> {
+  public async createPackage(userInput: ICreatePackage): Promise<ICreateJobResponse | ICallbackResponse> {
     const layer = await this.rasterCatalogManager.findLayer(userInput.dbId);
     const layerMetadata = layer.metadata;
     const { productId: resourceId, productVersion: version, footprint, productType } = layerMetadata;
     const { bbox, dbId, targetResolution, crs, priority, callbackURLs } = userInput;
     const zoomLevel = degreesPerPixelToZoomLevel(targetResolution);
 
+    const srcRes = layerMetadata.maxResolutionDeg as number;
+    const maxZoom = degreesPerPixelToZoomLevel(srcRes);
+    if (zoomLevel > maxZoom) {
+      throw new BadRequestError(`the requested requested resolution ${targetResolution} is larger then then product resolution ${srcRes}`);
+    }
+
+    const sanitizedBbox = this.sanitizeBbox(bbox, footprint as Polygon | MultiPolygon, zoomLevel);
+    if (sanitizedBbox === null) {
+      throw new BadRequestError(`requested bbox has no intersection with requested layer`);
+    }
+
     const dupParams: JobDuplicationParams = {
       resourceId: resourceId as string,
       version: version as string,
       dbId,
       zoomLevel,
-      bbox,
+      sanitizedBbox,
       crs: crs ?? DEFAULT_CRS,
     };
+    const callbacks = callbackURLs.map((url) => ({ url, bbox }));
 
-    const duplicationExist = await this.checkForDuplicate(dupParams, userInput.callbackURLs);
+    const duplicationExist = await this.checkForDuplicate(dupParams, callbacks);
     if (!duplicationExist) {
+      const batches: ITileRange[] = [];
+      for (let i = 0; i <= zoomLevel; i++) {
+        batches.push(bboxToTileRange(sanitizedBbox, i));
+      }
+      const separator = this.getSeparator();
+      const sources: IMapSource[] = [
+        {
+          path: this.generatePackageName(dbId, zoomLevel, sanitizedBbox), //gpkg path
```

Comment:
?

Reply:
geo package file name is required for the tile merger API.
it was previously generated in the imploder